lib/argonaut/bot.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Bot do
+  alias __MODULE__
+
+  @matchers [
+    {:reservation_action, ~r/(?<env>.*):(?<app>.*) (?<action>.*)/}
+  ]
+
+  @default_handler {:default_handler, nil}
+
+  def reply(message) when is_binary(message) do
+    {action, regex} = @matchers |> Enum.find(@default_handler, fn {_, re} ->
+      re |> Regex.match?(message)
```

Comment:
Nitpick, but should only use pipes if you're using more than one of them, otherwise, just use `Regex.match?(re, message)` in this case.